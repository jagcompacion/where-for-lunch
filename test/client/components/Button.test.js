import React from 'react';
import { shallow } from 'enzyme';
import Button from 'client/components/Button/Button';
import renderer from 'react-test-renderer';

test('With Enzyme, Button component will have disable property', () => {
  const wrapper = shallow(
    <Button disabled />,
  );
  const b = wrapper.find('button').props().disabled;
  expect(b).toBe(true);
});

test('With Enzyme, Button component will not have disable property', () => {
  const wrapper = shallow(
    <Button />,
  );
  const b = wrapper.find('button').props().disabled;
  expect(b).toBe(undefined);
});

test('With Jest snapshot, Button component renders with disable property', () => {
  const b = renderer
    .create(<Button disabled />)
    .toJSON();
  expect(b).toMatchSnapshot();
});
