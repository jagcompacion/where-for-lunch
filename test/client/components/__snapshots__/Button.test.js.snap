// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest snapshot, Button component renders with disable property 1`] = `
<button
  className="root undefined"
  disabled={true}
/>
`;
